File: c/chapitre_3/exo34.c
/*
Ecrire un programme qui saisit des noms de répertoires au clavier et copie
le répertoire courant dans tous ces répertoires. Le programme doit se poursuivre jusqu’à ce que
l’utilisateur demande de quitter le programme.
*/

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main()
{
    char *repertoire ;

    printf("Entrer le repertoire : ");
    scanf("%s",repertoire);

    execl("/usr/bin/cp" , "cp" , "-r" , getenv("PWD") , repertoire , NULL ) ;
    
    return 0 ;
}

File: python/chapitre_3/exo34.py
"""
Ecrire un programme qui saisit des noms de répertoires au clavier et copie
le répertoire courant dans tous ces répertoires. Le programme doit se poursuivre jusqu'à ce que
l'utilisateur demande de quitter le programme.
"""

import os
import shutil

def main():
    while True:
        repertoire = input("Entrer le repertoire : ")
        shutil.copytree(os.getcwd(), repertoire, dirs_exist_ok=True)

        rep = input("Voulez vous continuez à ajouter(o/n)")
        if rep != 'o':
            break

if __name__ == "__main__":
    main()